accepted/2025/PrunePackageReference-with-direct-PackageReference.md

```diff
@@ -0,0 +1,264 @@
+
+c. Multiple .NET frameworks, package specified for all frameworks.
+
+```csproj
+  <PropertyGroup>
+    <TargetFramework>net8.0;net9.0;net10.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Condition="'$(TargetFramework)' == 'net10.0'" Include="System.IO.Pipelines" Version="10.0.0-preview.3.25171.5" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net9.0'" Include="System.IO.Pipelines" Version="9.0.2" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net8.0'" Include="System.IO.Pipelines" Version="8.0.0" />
+  </ItemGroup>
+```
+
+In this situation, System.IO.Pipelines was added to the framework in .NET 9, so the .NET 10 and .NET 9 references are not necessary.
+There is a chance the customer may end up reference a System.IO.Pipelines version higher than the one available in the other frameworks, similarly to 2b.
+It is a lot less likely given that they're carefully targeting the latest version within the matching releases, but risky nonetheless.
+
+d. Multiple .NET frameworks, package is not specified in for frameworks
+
+```csproj
+  <PropertyGroup>
+    <TargetFramework>net8.0;net9.0;net10.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Condition="'$(TargetFramework)' == 'net10.0'" Include="System.IO.Pipelines" Version="10.0.0-preview.3.25171.5" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net9.0'" Include="System.IO.Pipelines" Version="9.0.2" />
+  </ItemGroup>
+```
+
+In this situation, System.IO.Pipelines was added to the framework in .NET 9, so the .NET 10 and .NET 9 references are not necessary.
+Given that the package is unnecessary in *every* framework it's been added in, we'd want the customer to remove both conditional PackageReference items.
+
+### How do we tell the customers we want them to remove a PackageReference
+
+We raise a NU1510 warning.
+
+For multi-targeted projects, the warning will be raised if at least one of the frameworks is .NET 10.
+We will raise an NU1510 when the PackageReference can be completely removed from the project file.
+In other words, when a package would be pruned for every target framework it is specified in, raise NU1510.
+
+In order to provide continuity, and minimize disruption that would be caused by raising NU1510 for existing projects, as per the [Breaking Change guidelines](https://github.com/dotnet/sdk/blob/main/documentation/project-docs/breaking-change-guidelines.md#tie-potentially-impactful-changes-to-the-target-tfm) NU1510 will only be raised for projects targeting net10, requiring an explicit customer action to get this diagnostic.
```

Comment:
> will only be raised for projects targeting net10

For policy like this, you may want to avoid hardcoding the versions in NuGet but instead let it be defined/overridden in MSBuild.

Reply:
The challenge is that this setting is a combination of the values of inner builds. 
I am assuming the question is more about people enabling it for more scenarios, correct?
I have a follow-up in https://github.com/NuGet/Home/issues/14126 that covers that.